### PR TITLE
Modularization

### PR DESCRIPTION
**Pull request by salceson**
Removed unnecessary stuff from core library.

### CODE REVIEW

File: src/main/scala/pl/edu/agh/iet/akka_tracing/TracedActor.scala [hunk 1]
```diff
@@ -6,25 +6,8 @@ import akka.actor.Actor
...
 case class MessageWrapper(id: UUID, msg: AnyRef)
 
-//trait DistributedStackTraceMessage {
-//  val stackTrace = Thread.currentThread().getStackTrace
-//}
-
-trait TracedActor {
-  actor: Actor =>
-
+trait TracedActor extends Actor {
```
Comment (mariusz89016) [inline src/main/scala/pl/edu/agh/iet/akka_tracing/TracedActor.scala:9]: Why does `TracedActor` extends `Actor`?

File: src/main/scala/pl/edu/agh/iet/akka_tracing/config/ConfigUtils.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+  }
+
+  implicit val stringConfigValueReader = new ConfigValueReader[String] {
+    override def fromConfig(config: Config, path: String): Option[String] = {
+      Try(config.getString(path)).toOption
+    }
+  }
+
+  implicit val configConfigValueReader = new ConfigValueReader[Config] {
```
Comment (mariusz89016) [inline src/main/scala/pl/edu/agh/iet/akka_tracing/config/ConfigUtils.scala:30]: `configConfigValueReader` :D

Comment (salceson) [reply src/main/scala/pl/edu/agh/iet/akka_tracing/config/ConfigUtils.scala:30]: Yeah, name is not good. Though to be consistent with others...

File: src/main/scala/pl/edu/agh/iet/akka_tracing/TracedActor.scala (hunk 1 above)
Comment (salceson) [reply src/main/scala/pl/edu/agh/iet/akka_tracing/TracedActor.scala:9]: This is actually a good question. The answer is: technical reason. I needed to have `aroundReceive` method here to be able to catch the call in aspect (even if the only thing it's calling is super). This method is private or package private so I could not have `actor.aroundReceive` here. So I needed to extend this trait to do this.

**Review by mariusz89016 — APPROVED** [review]

**Pull request merged**